### ISSUE TITLE
Save All doesn't gofmt all files when "go:formatOnSave" flag is true

### ISSUE THREAD

**Issue by lirao** [Microsoft/vscode-go#279]
When I save a bunch of files all at once with Save All, only the active file is formatted, and the rest are saved in their un-fmt'ed state. (Autosave is off)

**Comment by lukehoban** [comment]
I looked into this, and it appears it isn't possible to fix this with the current VS Code APIs.  

This is primarily blocked on https://github.com/Microsoft/vscode/issues/239, which would make sure the formatting could be applied before the files are saved to disk.  

But it is more directly blocked by seemingly not being able to get a `TextEditor` to use to apply a set of `TextEdit`s for a `TextDocument` provided to `vscode.workspace.onDidSaveTextDocument`.  See https://github.com/Microsoft/vscode-go/blob/master/src/goMain.ts#L186.  

/cc @egamma and @bpasero in case they have a suggestion for how this could be handled with the current VS Code API.

**Comment by bpasero** [comment]
I think this needs https://github.com/Microsoft/vscode/issues/239 and is a good scenario for having that as API.

**Comment by FrenchBen** [comment]
It'd be nice to also have:
`"go.fmtFlags": [],`

**Comment by ramya-rao-a** [comment]
`onWillSaveTextDocument` API is now available: https://code.visualstudio.com/updates/v1_6#_new-apis

The only catch is that there is a limitation on the time taken for the listeners of this event. 
There have been cases where the formatter in the GO extension seem to take a long time (multiple seconds) in which case, they will be eventually ignored: https://github.com/Microsoft/vscode/blob/a6435b1cbd1417b0d334ea9018f5371df198898f/src/vs/vscode.d.ts#L3980

**Comment by ramya-rao-a** [comment]
Closing in favor of #540 which will fix this issue as well

**Comment by ramya-rao-a** [comment]
With the work in #540 completed, the latest update to the Go extension (0.6.70) has the fix for this issue.
